views/charts.tmpl

```diff
@@ -0,0 +1,252 @@
+            tickets_price = [];
+            difficulty = [];
+            ticketsPrice.forEach(function(n){
+                tickets_price.push([new Date (n.time*1000), n.valuef]);
+                difficulty.push([new Date(n.time*1000), n.sizef]);
+            });
+        }
+
+        var supplyFunc = function (){
+            var supply = 0;
+            coin_supply = [];
+            coinSupply.forEach(function(n){
+                coin_supply.push([new Date(n.time*1000), (isNaN(n.valuef) ? 0 : n.valuef)]);
+            });
+        }
+        
+        var sizefunc = function (){
+            var size = 0;
+            block_size = [];
+            blockchain_size = [];
+            tx_per_block = [];
+            time_btw_blocks = [];
+            blockSize.forEach(function(n){
+                block_size.push([new Date(n.time*1000), n.size]);
+                size = size + n.size
+                blockchain_size.push([new Date(n.time*1000), size]);
+                tx_per_block.push([new Date(n.time*1000), n.count]);
+
+                if (isNaN(n.value)) { return; }
+                time_btw_blocks.push([n.value, (isNaN(n.valuef) ? 0 : n.valuef)]);
+            });
+        }
+
+        var txFunc = function (){
+            tx_per_day = [];
+            txPerDay.forEach(function(n){
+                tx_per_day.push([new Date(n.timestr), n.count]);
+            });
+        }
+
+        var poolFunc = function (){
+            pool_size = [];
+            pool_value = [];
+            poolInfo.forEach(function(n){
+                pool_size.push([new Date(n.time*1000), (isNaN(n.sizef) ? 0: n.sizef)]);
+                pool_value.push([new Date(n.time*1000), (isNaN(n.valuef) ? 0: n.valuef)]);
+            });
+        }
+
+        var blockFeeFunc = function (){
+            fee_per_block = [];
+            blockFee.forEach(function(n){
+                fee_per_block.push([n.count, (isNaN(n.sizef) ? 0: n.sizef)]);
+            }); 
+        }
+
+        var drawGraph = function (data,labelsVal, isDrawPoint, yLabel, xLabel, titleName, labelsMG, labelsMG2) {
+             g.updateOptions({ 
+                    'file': data,
+                    labels: labelsVal,
+                    drawPoints: isDrawPoint,
+                    ylabel: yLabel,
+                    xlabel: xLabel,
+                    labelsKMB: labelsMG,
+                    labelsKMG2: labelsMG2,
+                    title: titleName
+                });
+                safeStopTurbolinksProgress();
+            }
+
+        function resolvePromise(func) {
+        return new Promise(resolve => {
+            setTimeout(() => {
+            func()
+            resolve('resolved');
+            }, 2000);
+        });
+        }
+
+        var plotGraph = async function (value){
+            switch(value){
+                case "1": // price graph
+                    if (tickets_price === undefined) { await resolvePromise(ticketsFunc); }
```

Comment:
What is the purpose of using asynchronous primitives here? Or added any sort of delay at all?
Why not just execute the transcoding step immediately like so?
```
if (tickets_price === undefined) { ticketsFunc() }
```